homeassistant/components/neevo/__init__.py

```diff
@@ -60,25 +56,30 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     except (ClientError, GenericHTTPError, InvalidResponseFormat) as err:
         raise ConfigEntryNotReady from err
 
-    hass.data[DOMAIN][API_CLIENT][config_entry.entry_id] = api
-    hass.data[DOMAIN][TANKS][config_entry.entry_id] = tanks
-
-    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
-
-    def update_published():
-        """Handle a push update."""
-        dispatcher_send(hass, PUSH_UPDATE)
+    async def fetch_update():
+        """Fetch the latest changes from the API."""
+        try:
+            return await api.get_tanks_info()
+        except PyNeeVoError as err:
+            raise UpdateFailed(err) from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="Neevo",
+        update_method=fetch_update,
+        update_interval=timedelta(minutes=INTERVAL),
+    )
 
-    for _tank in tanks.values():
-        _tank.set_update_callback(update_published)
+    await coordinator.async_config_entry_first_refresh()
 
-    async def fetch_update(now):
-        """Fetch the latest changes from the API."""
-        await api.refresh_tanks()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = {
+        "coordinator": coordinator,
```

Comment:
You'll be better typo-proofed by putting "coordinator" in your const.py, like you used to have with API_CLIENT.